### PR TITLE
Fixing various bugs

### PR DESCRIPTION
**Pull request by JTPetter**
- Linearity Study: Added sensible error message when only one observation per part.
- Individual Charts: Added sensible error message when moving range length was larger than one of the subgroups.
- Individual Charts: Fix crashes of IMR chart when moving range length is equal to data/stage length.
- Fix gauge r&R non-rep breaking when less than two measurements per part per operator
- Fix test-retest breaking when using long format data.
- Fix Time-weighted charts not working if more than one variable is added.
- Excluded g chart from the module, always breaks and calculations are unclear.
- Made CumSum Chart checked by default, so analysis does not open up completely empty.

### CODE REVIEW

**Comment by JTPetter** [comment]
Interestingly the unit tests worked for [edd893a](https://github.com/jasp-stats/jaspQualityControl/pull/254/commits/edd893a9fe577faf8a3d9b92a9965bb739cf1340), but not anymore for [ce2b4bd](https://github.com/jasp-stats/jaspQualityControl/pull/254/commits/ce2b4bde4e5ec09b1c2f1048cfd76e78cacb9bba), although I really don't see what could have caused them to fail in this change?

**Comment by JTPetter** [comment]
... and somehow my new commits to https://github.com/JTPetter/jaspQualityControl/tree/fixingVariousBugs don't show up in this PR anymore?

File: R/msaGaugeRRnonrep.R [hunk 1]
```diff
@@ -362,12 +362,11 @@ msaGaugeRRnonrep <- function(jaspResults, dataset, options, ...) {
 }
 
 .reshapeToWide <- function(dataset, measurements, parts, operators) {
-  nreplicates       <- as.vector(table(dataset[parts])[1])
-  nparts <- length(unique(dataset[[parts]]))
   dataset <- dataset[order(dataset[parts]),]
-  dataset <- cbind(dataset, data.frame(index = rep(1:nreplicates, nparts)))
+  index <- sequence(plyr::count(dataset, vars = c(parts, operators))$freq)
```
Comment (vandenman) [inline R/msaGaugeRRnonrep.R:366]: Can this use `dplyr::count` instead? `plyr` is retired so it's probably better to use `dplyr` or another package instead.

File: R/msaGaugeLinearity.R [hunk 2]
```diff
@@ -85,17 +85,25 @@ msaGaugeLinearity <- function(jaspResults, dataset, options, ...) {
 
   if (ready) {
 
+    # Error conditions
     if (length(dataset[[measurements]]) < 2){
-      table2$setError(gettextf("T-Test requires more than 1 measurement. %i valid measurement(s) detected in %s.", length(dataset[[measurements]]), measurements))
+      table2$setError(gettextf("T-Test requires more than 1 measurement in total. %i valid measurement(s) detected in %s.", length(dataset[[measurements]]), measurements))
```
Comment (vandenman) [inline R/msaGaugeLinearity.R:90]:
```suggestion
      table2$setError(gettextf("T-Test requires more than 1 measurement in total. %1$i valid measurement(s) detected in %2$s.", length(dataset[[measurements]]), measurements))
```

File: R/msaGaugeLinearity.R [hunk 3]
```diff
@@ -85,17 +85,25 @@ msaGaugeLinearity <- function(jaspResults, dataset, options, ...) {
...
       return(table2)
-    }
-    else if (length(unique(dataset[[standards]])) != length(unique(dataset[[parts]]))) {
+    } else if (length(unique(dataset[[standards]])) != length(unique(dataset[[parts]]))) {
       table2$setError(gettextf("Every unique part must have one corresponding reference value. %i reference values were found for %s unique parts.", length(unique(dataset[[standards]])), length(unique(dataset[[parts]]))))
```
Comment (vandenman) [inline R/msaGaugeLinearity.R:93]: same as the above, use `%$1x` and `%$2x`.

**Review by vandenman — CHANGES_REQUESTED** [review]

**Comment by JTPetter** [comment]
@vandenman Thank you, I addressed your comments.

**Pull request merged**